rasa/__main__.py

```diff
@@ -68,6 +69,8 @@ def main() -> None:
     )
     set_log_level(log_level)
 
+    tf_env.setup_tf_environment()
```

Comment:
Isn't `main()` always run for all rasa commands(e.g. `train`,`test`,etc.)
cc @wochinge

Reply:
True, we missed one case @dakshvar22 And that's when Rasa X is run in local mode because then we start a separate process (the main is still called, but since we start a separate process it will not be applied)

Reply:
@wochinge But in that case, the separate process will have the appropriate tensorflow environment settings which is good assuming that model training happens as part of that separate process?

Reply:
The env variables will be part oft the new access, but you have to re-apply them to the tf.config

Reply:
Also I don't think local Rasa X needs to support fancy TensorFlow stuff (not scalable anyway)

Reply:
Also doesn’t work if someone calls agent.train directly from their code

Reply:
In that case they can / should set the tf config themselves

Reply:
Yes, in the case of external call to `agent.train`, I think they should setup the environment themselves. I can put this as a note in the docs.
@tmbo What do you suggest for the case of local Rasa X mode?

Reply:
what happens if someone calls `agent.train` but does not call `tf_env.setup_tf_environment()` before, does it still work?

Reply:
No, this wouldn't work. Tensorflow changed the way how its environment is setup considerably. They require a global process level configuration now. So, if someone calls `agent.train` from their external code it could also mean they have other tensorflow models running outside of Rasa but as part of the same process. In that case, I would propose that the configuration is setup by the user themselves. I agree, we should put this case as a note in the docs where the tensorflow environment setup is explained.

Reply:
@tmbo  Is there a better place within the codebase where we can invoke this function to ensure all cases are covered?

Reply:
is there anyway for us to know if the environment is already properly set up? i.e. what happens uf `setup_tf_environment` is called multiple times?

If that is not an issue, it should be called everywhere where it is needed.

Reply:
@tmbo Umm I don't think so. Can't make multiple calls to setup the GPU again once it is setup which is done here - https://github.com/RasaHQ/rasa/pull/5266/files#diff-02b386eef06397c23a266513924e68faR18-R40 as part of `setup_tf_environment` function

Reply:
More so, it would result in a `RuntimeError` if the tensorflow runtime is initialized once already.